lib/http/connection.rb

```diff
@@ -86,10 +86,16 @@ def send_request(req)
     def readpartial(size = BUFFER_SIZE)
       return unless @pending_response
 
-      finished = (read_more(size) == :eof) || @parser.finished?
-      chunk    = @parser.chunk
-
-      finish_response if finished
+      if @parser.chunk && @parser.chunk.bytesize <= size
+        chunk = @parser.chunk
+      elsif @parser.chunk
+        chunk = @parser.chunk.byteslice(0, size)
+        @parser.chunk[0, size] = ""
```

Comment:
This line is not correct, that would make `@parser.chunk` an empty string when there is no content, and then the `if @parser.chunk` condition would always hold.